pkg/apis/resource/validation/validation.go

```diff
@@ -266,6 +269,21 @@ func validateResourceClaimStatusUpdate(status, oldStatus *resource.ResourceClaim
 		func(consumer resource.ResourceClaimConsumerReference) (types.UID, string) { return consumer.UID, "uid" },
 		fldPath.Child("reservedFor"))...)
 
+	if utilfeature.DefaultFeatureGate.Enabled(features.DRAResourceClaimDeviceStatus) {
+		var allocatedDevices sets.Set[structured.DeviceID]
+		if status.Allocation != nil {
+			allocatedDevices = gatherAllocatedDevices(&status.Allocation.Devices)
+		}
```

Comment:
related to this https://github.com/kubernetes/kubernetes/pull/128240/files#r1823538165

I don't think the apiserver should participate here and just represent and validate the state created by the different/controllers and drivers by setting these fields

Reply:
hmm, I see, I didn't understand this until I read this code sorry, so this validates that the status belongs to an existing device ...